src/components/progressCircular/progress-circular.js

```diff
@@ -79,19 +84,79 @@ function MdProgressCircularDirective($mdConstant, $mdTheming) {
 
   function postLink(scope, element, attr) {
     $mdTheming(element);
+
     var circle = element[0];
+    var spinnerWrapper =  angular.element(element.children()[0]);
+
+    var lastMode, toVendorCSS = $mdUtil.dom.animator.toCss;
+
+    // Update size/scaling of the progress indicator
+    // Watch the "value" and "md-mode" attributes
+
     circle.style[$mdConstant.CSS.TRANSFORM] = 'scale(' + getDiameterRatio() + ')';
 
     attr.$observe('value', function(value) {
       var percentValue = clamp(value);
       element.attr('aria-valuenow', percentValue);
+
+      if (attr.mdMode == "determinate") {
+        animateIndicator(percentValue);
+      }
     });
 
-    var spinnerWrapper =  angular.element(element.children()[0]);
     attr.$observe('mdMode',function(mode){
-      spinnerWrapper[mode ? 'removeClass' : 'addClass']('ng-hide');
+      switch( mode ) {
+        case MODE_DETERMINATE:
+        case MODE_INDETERMINATE:
+          spinnerWrapper.removeClass('ng-hide');
+
+          // Inject class selector instead of attribute selector
+          // (@see layout.js changes for IE performance issues)
+
+          if ( lastMode ) spinnerWrapper.removeClass(lastMode);
```

Comment:
Only if `mode="indeterminate || mode == "determinate` will we show the spinner container and its contents.